koji_containerbuild/plugins/builder_containerbuild.py

```diff
@@ -945,3 +954,271 @@ def handler(self, src, target, opts=None):
+        build_identifier = build_nvr or build_id
+
+        koji_build = self.session.getBuild(build_identifier)
+        if not koji_build:
+            raise koji.BuildError("specified source build '%s' doesn't exist" % build_identifier)
+
+        if build_id and (build_id != koji_build['build_id']):
+            err_msg = (
+                'koji_build_id {} does not match koji_build_nvr {} with id {}. '
+                'When specifying both an id and an nvr, they should point to the same image build'
+                .format(build_id, build_nvr, koji_build['build_id'])
+                )
+            raise koji.BuildError(err_msg)
+
+        if not build_id:
+            build_id = koji_build['build_id']
+        if not build_nvr:
+            build_nvr = koji_build['nvr']
+        component = "{}-source" % koji_build['package_name']
+
+        return component, build_id, build_nvr
+
+    def handler(self, target, opts=None):
+        jsonschema.validate([target, opts], self.PARAMS_SCHEMA)
+
+        if opts is None:
+            opts = {}
+        self.opts = opts
+
+        self.event_id = self.session.getLastEvent()['id']
+        target_info = self.session.getBuildTarget(target, event=self.event_id)
+        if not target_info:
+            raise koji.BuildError("Target `%s` not found" % target)
+
+        component, build_id, build_nvr = self.get_source_build_info(opts.get('koji_build_id'),
+                                                                    opts.get('koji_build_nvr'))
+        # scratch builds do not get imported, and consequently not tagged
+        if not self.opts.get('scratch'):
+            self.check_whitelist(component, target_info)
+
+        try:
+            results = self.runBuilds(target_info,
+                                     scratch=opts.get('scratch', False),
+                                     component=component,
+                                     koji_build_id=build_id,
+                                     koji_build_nvr=build_nvr,
+                                     signing_intent=opts.get('signing_intent', None),
+                                     )
+            all_repositories = []
+            all_koji_builds = []
+
+            for result in results:
+                try:
+                    repository = result.get('repositories')
+                    all_repositories.extend(repository)
+                except Exception as error:
+                    self.logger.error("Failed to merge list of repositories "
+                                      "%r. Reason (%s): %s", repository,
+                                      type(error), error)
+                koji_build_id = result.get('koji_build_id')
+                if koji_build_id:
+                    all_koji_builds.append(koji_build_id)
+
+        except (SystemExit, ServerExit, KeyboardInterrupt):
+            # we do not trap these
+            raise
+        except:
+            # reraise the exception
+            raise
```

Comment:
Confused about why we have a `try` block at all here!

Reply:
yeah seems useless since it will reraise anyway